src/main/java/cz/tstrecha/timetracker/service/impl/UserServiceImpl.java

```diff
@@ -134,6 +127,16 @@ public RelationshipDTO updateRelationship(RelationshipCreateUpdateRequestDTO req
         return relationshipMapper.toDTOFromReceiving(relation);
     }
 
+    public String hasPermissionToChangeContext(Long id, LoggedUser loggedUser) {
+        var relationship = userRelationshipRepository.findByFromAndTo(loggedUser.getUserEntity(),
```

Comment:
Its not best practice to have so nested calling

But theres another problem -> UserContext has every possible user relationship, therefor every call to change context shouldnt always look for active relationship and then map, it should:

- Verify user can login as this user from context
- Find this relationship
- Create token a return it